modules/managed-identity/user-assigned-identity/.bicep/nested_rbac.bicep

```diff
@@ -0,0 +1,55 @@
+@sys.description('Required. The IDs of the prinicpals to assign to role to')
+param principalIds array
+
+@sys.description('Required. The name of the role to assign. If it cannot be found you can specify the role definition ID instead')
+param roleDefinitionIdOrName string
+
+@sys.description('Required. The resource ID of the resource to apply the role assignment to')
+param resourceId string
+
+@sys.description('Optional. The principal type of the assigned principal ID.')
+@allowed([
+  'ServicePrincipal'
+  'Group'
+  'User'
+  'ForeignGroup'
+  'Device'
+  ''
+])
+param principalType string = ''
+
+@sys.description('Optional. Description of role assignment')
+param description string = ''
+
+var builtInRoleNames = {
```

Comment:
one thing about providing this list (in every module) is that users won't know what's available in any given module, with searching the code.  If we follow this pattern, it feels like it has to work everywhere (in every roleAssignment module) or we should require the resource name and provide a lookup function in ARM.

cc: @alex-frankel